Observer/InStorePickupHandler.php

```diff
@@ -0,0 +1,221 @@
+
+                $order->setShippingDescription($description);
+            } else {
+                $this->inventoryPickupLocationTableManager->removeRowByOrderId((int)$order->getEntityId());
+
+                $quoteShippingAddress = $quote->getShippingAddress();
+                $this->inventoryPickupLocationTableManager->removeRowByQuoteAddressId($quoteShippingAddress->getData('address_id') ?? 0);
+            }
+        } catch (LocalizedException $localizedException) {
+            $this->logger->alert($localizedException->getLogMessage());
+        }
+    }
+
+    /**
+     * @throws NoSuchEntityException
+     */
+    protected function changeQuoteAddressToPickupLocation(Quote $quote, string $pickupLocationCode): void
+    {
+        $address = $quote->getShippingAddress();
+        if ($pickupLocationCode && $address->getShippingMethod() === InStorePickup::DELIVERY_METHOD) {
+            $this->setAddressPickupLocation->execute($address, $pickupLocationCode);
+            $quote->setShippingAddress(
+                $this->getShippingAddressBySourceCodeAndOriginalAddress->execute($pickupLocationCode, $address)
+            );
+        }
+        $this->cartRepository->save($quote);
+    }
+
+    /**
+     * @throws NoSuchEntityException
+     */
+    protected function changeOrderAddressToPickupLocation(Order $order, Quote $quote, string $pickupLocationCode): void
+    {
+        $quoteShippingAddress = $quote->getShippingAddress();
+
+        try {
+            $source = $this->sourceRepository->get($pickupLocationCode);
+            $additionalAddressInformation = [
+                'address_type' => 'shipping',
+                'firstname'    => $source->getData('frontend_name') ?? $source->getName(),
+                'email'        => $source->getEmail(),
+                'telephone'    => $source->getPhone()
+            ];
+        } catch (NoSuchEntityException $exception) {
+            $additionalAddressInformation = [
+                'address_type' => 'shipping',
+                'email'        => $quote->getCustomerEmail()
+            ];
+        }
+
+        $additionalAddressInformation['middlename'] = '';
+        $additionalAddressInformation['lastname'] = '';
+
+        $shippingAddress = $this->quoteAddressToOrderAddress->convert(
+            $quoteShippingAddress,
+            $additionalAddressInformation
+        );
+
+        $shippingAddress->setData('quote_address_id', $quote->getShippingAddress()->getId());
+        $orderShippingAddress = $order->getShippingAddress();
+
+        if (!is_null($orderShippingAddress) && !empty($shippingAddress->getData())) {
+            $orderShippingAddress->addData($shippingAddress->getData());
+            $this->orderAddressRepository->save($orderShippingAddress);
+        }
+    }
+
+    protected function setOrderExtensionAttribute(Order $order, string $pickupLocationCode = ''): void
```

Comment:
`setPickupLocationCodeInOrder` will be more accurate method name

Reply:
Chanched to setPickupLocationCodeInOrderExtensionAttribute. Since it will be attached to the order already in the plugin later